### PR TITLE
Use enums for remaining metrics used by metric sets

### PR DESCRIPTION
**Pull request by yngveaasheim**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@olaaun or @kkraune or @gjoranv Please review

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/admin/monitoring/VespaMetricSet.java [hunk 1]
```diff
@@ -55,75 +60,71 @@ private static Set<Metric> getVespaMetrics() {
...
-        addMetric(metrics, "vds.server.network.tls-connections-broken");
-        addMetric(metrics, "vds.server.network.failed-tls-config-reloads");
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_TLS_HANDSHAKES_FAILED.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_PEER_AUTHORIZATION_FAILURES.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_CLIENT_TLS_CONNECTIONS_ESTABLISHED.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_SERVER_TLS_CONNECTIONS_ESTABLISHED.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_CLIENT_INSECURE_CONNECTIONS_ESTABLISHED.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_SERVER_INSECURE_CONNECTIONS_ESTABLISHED.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_TLS_CONNECTIONS_BROKEN.count());
+        addMetric(metrics, StorageMetrics.VDS_SERVER_NETWORK_FAILED_TLS_CONFIG_RELOADS.count());
```
Comment (olaaun) [inline config-model/src/main/java/com/yahoo/vespa/model/admin/monitoring/VespaMetricSet.java:95]: Is the change from base name to `.count` intended?

Comment (yngveaasheim) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/monitoring/VespaMetricSet.java:95]: Yes, intended. Should have been stated in the commit message, obviously. These have never been propagated to Yamas for this reason.

**Review by gjoranv — APPROVED** [review]

File: config-model/src/main/java/com/yahoo/vespa/model/admin/monitoring/VespaMetricSet.java (hunk 1 above)
Comment (olaaun) [reply config-model/src/main/java/com/yahoo/vespa/model/admin/monitoring/VespaMetricSet.java:95]: 👍

**Review by olaaun — APPROVED** [review]

File: container-core/src/main/java/com/yahoo/metrics/ConfigServerMetrics.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+/**
+ * @author yngveaasheim
+ */
+public enum ConfigServerMetrics implements VespaMetrics {
+
+    CONFIGSERVER_REQUESTS("configserver.requests", Unit.REQUEST, "Number of requests processed"),
+    CONFIGSERVER_FAILED_REQUESTS("configserver.failedRequests", Unit.REQUEST, "Number of requests that failed"),
+    CONFIGSERVER_LATENCY("configserver.latency", Unit.MILLISECOND, "Time to complete requests"),
+    CONFIGSERVER_CACHE_CONFIG_ELEMS("configserver.cacheConfigElems", Unit.ITEM, "Time to complete requests"),
+    CONFIGSERVER_CACHE_CHECKSUM_ELEMS("", Unit.ITEM, "Number of checksum elements in the cache"),
```
Comment (baldersheim) [inline container-core/src/main/java/com/yahoo/metrics/ConfigServerMetrics.java:12]: Missing name for metric. Fixed in #26293

File: container-core/src/main/java/com/yahoo/metrics/ConfigServerMetrics.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    // ZooKeeper related metrics
+    CONFIGSERVER_ZK_CONNECTIONS_LOST("configserver.zkConnectionLost", Unit.CONNECTION, "Number of ZooKeeper connections lost"),
+    CONFIGSERVER_ZK_RECONNECTED("configserver.zkReconnected", Unit.CONNECTION, "Number of ZooKeeper reconnections"),
+    CONFIGSERVER_ZK_CONNECTED("configserver.zkConnected", Unit.NODE, "Number of ZooKeeper nodes connected"),
+    CONFIGSERVER_ZK_SUSPENDED("configserver.zkSuspended", Unit.NODE, "Number of ZooKeeper nodes suspended"),
+    CONFIGSERVER_ZK_Z_NODES("configserver.zkZNodes", Unit.NODE, "Number of ZooKeeper nodes present"),
+    CONFIGSERVER_ZK_AVG_LATENCY("configserver.zkAvgLatency", Unit.MILLISECOND, "Average latency for ZooKeeper requests"), // TODO: Confirm metric name
+    CONFIGSERVER_ZK_MAX_LATENCY("configserver.zkMaxLatency", Unit.MILLISECOND, "Max latency for ZooKeeper requests"),
+    CONFIGSERVER_ZK_CONNECTIONS("configserver.zkConnections", Unit.CONNECTION, "Number of ZooKeeper connections"),
+    CONFIGSERVER_ZK_OUTSTANDING_REQUESTS("configserver.zkOutstandingRequests", Unit.REQUEST, "Number of ZooKeeper requests in flight");
```
Comment (baldersheim) [inline container-core/src/main/java/com/yahoo/metrics/ConfigServerMetrics.java:36]: All metrics here have the same "configserver." prefix, and the class is ConfigServerMetrics. I would drop the CONFIGSERVER_ prefix for the members.
less is more.

**Pull request merged**